ilastik/experimental/parser/_base.py

```diff
@@ -1,137 +1,41 @@
-import json
-import pickle
-from typing import Any, List, Optional
+from typing import Optional
 
 import h5py
-import numpy
 
 from . import types
 
-PIXEL_CLASSIFICATION = b"Pixel Classification"
 
+class PixelClassificationProject:
+    ILP_WORKFLOW_NAME = b"Pixel Classification"
 
-class IlastikProject:
-    def __init__(self, path: str, mode: str = "r") -> None:
-        if mode != "r":
-            raise NotImplementedError(f"mode: {mode}")
-
-        self.__mode = mode
-        self.__path = path
-        self.__file = None
-
-    def __enter__(self):
-        self.__file = h5py.File(self.__path, mode=self.__mode)
-        return _create_project_wrap(self.__file)
-
-    def __exit__(self, exc_type, exc, tb):
-        if self.__file:
-            self.__file.close()
-
-
-WORKFLOW_KEY = "workflowName"
-
-
-def _create_project_wrap(hdf5_file):
-    type_ = hdf5_file[WORKFLOW_KEY][()]
-    if type_ == PIXEL_CLASSIFICATION:
-        return _PixelClassProjectImpl(hdf5_file)
-
-    raise NotImplementedError(f"Unknown project type {type_}")
-
-
-class _Keys:
-    INPUT_DATA = "Input Data"
-    INPUT_DATA_INFOS = "infos"
-    INPUT_DATA_RAW = "Raw Data"
-    INPUT_DATA_AXIS_TAGS = "axistags"
-    INPUT_DATA_SHAPE = "shape"
-
-    FEATURES = "FeatureSelections"
-    FEATURES_IDS = "FeatureIds"
-    FEATURES_SCALES = "Scales"
-    FEATURES_COMPUTE_IN_2D = "ComputeIn2d"
-    FEATURES_SELECTION_MATRIX = "SelectionMatrix"
-    PIXEL_CLASSIFICATION = "PixelClassification"
-    PIXEL_CLASSIFICATION_TYPE = "pickled_type"
-    PIXEL_CLASSIFICATION_FACTORY = "ClassifierFactory"
-    PIXEL_CLASSIFICATION_FORESTS = "ClassifierForests"
-    LABEL_NAMES = "LabelNames"
+    def __init__(self, hdf5_file: h5py.File) -> None:
+        class ILPkeys(types.StrEnum):
```

Comment:
What's the reason for using enum classes instead of local variables or class attributes?

Reply:
convenience... with this one can 

```Python
if any(key not in hdf5_file for key in ILPkeys):
...
```

Reply:
In this case I think it's better to avoid `Enum` because values like `ILPKeys.WORKFLOW` have types that are different from their values (e.g. `ILPKeys.WORKFLOW` is not `str`), which could potentially break something downstream.

One could cook up a simple metaclass that does approximately the same but preserves actual types (really this just adds convenient iteration, nothing more):

```python
class BagMeta(type):
    def __iter__(cls):
        # Ignore built-in dunders
        return iter(v for k, v in cls.__dict__.items() if not k.startswith("_"))


# Just to make inheritance more convenient.
class bag(metaclass=BagMeta):
    pass


class values(bag):
    spam = "Lorem Ipsum"
    eggs = 42
    # Bonus feature: hide something from iteration
    _ham = "hidden!"


print(list(values))
# ['Lorem Ipsum', 42]
```

Reply:
Note that it's possible to get the actual value from enum member with `ILPKeys.WORKFLOW.value`, but it's an additional attribute access.

Reply:
hmm, I don't know, with `StrEnum` you get `isinstance(`ILPKeys.WORKFLOW`, "str") is True`...

Reply:
from what I read in the docs, it's pretty much intended use: https://docs.python.org/3.9/library/enum.html#others

Reply:
I think this is the intended use if you use enum members as unique symbols and maybe sometimes access their values via `MyEnum.MY_MEMBER.value`, but enum members and their values are still different:

```python
import enum

class StrEnum(str, enum.Enum):
   pass

class my(StrEnum):
   a = 'value of a'
   b = 'value of b'

print(list(my))
# [<my.a: 'value of a'>, <my.b: 'value of b'>]
```

But I'm not sure to what extent these differences matter in practice...

Reply:
gonna stick with enums, they're only used in `from_project_file`, their behavior is quite predictable there, without the need for any custom meta classes...